src/sync_template.sh

```diff
@@ -53,6 +54,13 @@ echo "::endgroup::"
 
 echo "::group::commit and push changes"
 git add .
+
+if [ -r ${TEMPLATE_SYNC_IGNORE_FILE_NAME} ]
+then
+  echo "::debug::unstage files from template sync ignore"
+  git reset `cat ${TEMPLATE_SYNC_IGNORE_FILE_NAME}`
```

Comment:
This should surely reset the files given as `glob` inside of the ignore file.
Am currently not 100% sure if this works for new files added to the template repository (which are present in the ignore file) when using `--allow-unrelated-histories` flag.
Maybe a combination with [git-clean](https://git-scm.com/docs/git-clean) is doing the trick.

Reply:
Thank you for the input! I applied your suggestion and added some cleanup steps after resetting the files. I am now calling `git clean -df` to remove untracked files and `git checkout -- .` to discard unstaged changes.

Reply:
Thanks a lot @jg-rivera . I am good to go with your PR :+1: 
Just one more (small) thing I want to discuss.
There are small [shell-check](https://www.shellcheck.net/) issues [SC2006](https://github.com/koalaman/shellcheck/wiki/SC2006) and [SC2046](https://github.com/koalaman/shellcheck/wiki/SC2046)?

Highly appreciated to discuss those :)

Am happy to merge that in afterwards.

Reply:
Thank you for the thorough review. I learned a lot of new things! ✨

Complying with SC2006 and SC2046 was a bit tricky. On this [commit](https://github.com/AndreasAugustin/actions-template-sync/pull/88/commits/180b5e88738e60b96fd95fa3d1fe20038f6d6608), I originally used a `while` loop but that terribly spammed the logs and it seemed inefficient. 

To effectively comply, I discovered and used the `--pathspec-from-file=<file>` option instead for [git-reset](https://git-scm.com/docs/git-reset).

Reply:
> Thank you for the thorough review. I learned a lot of new things! 
> 
> Complying with SC2006 and SC2046 was a bit tricky. On this [commit](https://github.com/AndreasAugustin/actions-template-sync/pull/88/commits/180b5e88738e60b96fd95fa3d1fe20038f6d6608), I originally used a `while` loop but `git reset` terribly spammed the logs and it seemed inefficient.
> 
> To effectively comply, I discovered and used the `--pathspec-from-file=<file>` option instead for [git-reset](https://git-scm.com/docs/git-reset).

I think

```bash
git reset $(cat "${TEMPLATE_SYNC_IGNORE_FILE_NAME}")
```

would have been an option, but I like your solution with `--pathspec-from-file=<file>` way more :)

I will think about a way how to add some unit tests. Am not :100: % sure yet how to do it. Maybe [go-git](https://github.com/go-git/go-git) is an option #92 

Nevertheless this has nothing to do with the current implementation. Thanks a lot!

Happy coding :rocket: